### DISCUSSION TITLE
Create and set custom html dom attributes

### DISCUSSION THREAD

**Discussion by eninthala** [ether/etherpad-lite#5676] [Q&A]
Hi,

I want to create custom html tags inside the span tag given by etherpad by default.

The question is: How can I programmatically write another attribute to this default span?

For instance, we already have:
`<span class>Hello world!</span>`

But I want to do this:
`<span data-id=12345 data-old="test" class>Hello world!</span>`

I need to have data-id & data-old tags as default and I need to set these values from client side.

As per the suggestion [here](https://github.com/ether/etherpad-lite/issues/643#issuecomment-5171633), I used the aceCreateDomLine client hook. I modified the domline.js as follows:

```
html.push('<span data-id="DATA_ID"  data-old="DATA_OLD" class="', Security.escapeHTMLAttribute(cls || ''), '">',
extraOpenTags, perTextNodeProcess(Security.escapeHTML(txt)), extraCloseTags, '</span>');
```

Instead of hardcoding static values I need to send the DATA_ID and DATA_OLD from client side.

Please let me know your thoughts.

Thanks in advance!

**Comment by SamTV12345** [comment]
If this is JavaScript shouldn't you be able to replace ' with ` and insert the desired values in it?
